lib/internal/repl.js

```diff
@@ -121,22 +121,43 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
     if (data) {
       repl.history = data.split(/[\n\r]+/, repl.historySize);
     } else if (oldHistoryPath) {
-      // Grab data from the older pre-v3.0 JSON NODE_REPL_HISTORY_FILE format.
-      repl._writeToOutput(
+      if (oldHistoryPath === historyPath) {
```

Comment:
It could but that we give rise to `else if` branches with repeated checks. It would end up looking something like the following:

```javascript
if (data) {
  repl.history = data.split(/[\n\r]+/, repl.historySize);
} else if (oldHistoryPath && oldHistoryPath === historyPath) {
  // If pre-v3.0, the user had set NODE_REPL_HISTORY_FILE to
  // ~/.node_repl_history. Warn the user about it and proceed.
  ...
} else if (oldHistoryPath && oldHistoryPath !== historyPath) {
  // Grab data from the older pre-v3.0 JSON NODE_REPL_HISTORY_FILE format.
  repl._writeToOutput(
    '\nConverting old JSON repl history to line-separated history.\n' +
    `The new repl history file can be found at ${historyPath}.\n`);
  repl._refreshLine();
  ...
}
```